Cupcake.cpp

```diff
@@ -0,0 +1,142 @@
+#include "Cupcake.h"
+#include "Player.h"
+
+using namespace sf;
+//Constructor - Destructor
+Cupcake::Cupcake() 
+{
+	srand((int)time(0));
+
+	//Randomize cupcake speed
+	m_cupcakeSpeed = (rand() % 2) + 8;
+	m_maxCupcakes = 5;
+
+	//Initialize bool
+	m_deleted = false;
+	
+}
+
+Cupcake::~Cupcake()
+{
+
+}
+
+
+void Cupcake::update(sf::RenderWindow& w)
+{
+		
+		//Update and reset the timer for spawning cupcakes and moving them
+		m_cupcakeSpawnTimer += .3f;
+
+		//Update the cupcakes as per last frame
+		removeCupcake();
+		
+
+		//spawn and move cupcakes
+		if (m_cupcakeSpawnTimer > m_cupcakeSpawnTimerMax)

```

Comment:
if I leave m_cupcakeSpawnTimer > 1.0 f the speed at which the cupcakes fall is huge (can barely see them on the screen). The number I found works best here is at 50.f. Not sure if this is because of what's happening in this function or in the others. To investigate further...